Matrix.ConsoleApp/Matrix.cs

```diff
@@ -0,0 +1,122 @@
+/// [(i==j), (i!=j), (i!=j)] (X, 0, 0) is to index this 'X'
+/// [(i!=j), (i==j), (i!=j)] (0, X, 0)
+/// [(i!=j), (i!=j), (i==j)] (0, 0, X)
+/// </summary>
+namespace Matrix.ConsoleApp
+{
+    internal class Matrix
+    {
+        //1.
+        public int[] DiagonalMatrixArray;
+        //2.
+        public int Size => _size;
+        private readonly int _size;
+        //the purpose of the indexer
+        //https://www.geeksforgeeks.org/c-sharp-multidimensional-indexers/ 
+        //4. The two indices as a multidimensional indexer
+
+        public int this[int i, int j]
+        {
+            get
+            {                
+                if (i < 0 && j < 0 || i >= _size && j >= _size)
+                {
+                    return 0;
+                }
+                else if (i != j)
+                {
+                    return 0;
+                }
+                else if (i == j)
+                { 
+                    return DiagonalMatrixArray[i];
+                }
+                else return 0;
+            }
+            set => DiagonalMatrixArray[i] = value;
+        }
+        //passing a a single-dimentional array of params as in ms.docs
+        //3.
+        //https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/arrays/passing-arrays-as-arguments
+        //https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/params
+        //https://www.c-sharpcorner.com/UploadFile/c63ec5/use-params-keyword-in-C-Sharp/
+        public Matrix(params int[] diagonalMatrixArray)
+        {
+            DiagonalMatrixArray = diagonalMatrixArray;
+            _size = diagonalMatrixArray.Length;
+
+            //validating if params element is null
+            //https://stackoverflow.com/questions/6584131/c-sharp-is-it-possible-to-have-null-params @Joshua Rodgers answer
+            if (diagonalMatrixArray == null)
+            {
+                _size = 0;
+            }
+            else
+            {
+                //a loop for creating a matrix, iterating through rows and columns:
+                for (int row = 0; row < _size; row++)
+                {
+                    Console.WriteLine();
+                    for (int column = 0; column < _size; column++)
+                    {
+                        Console.Write($"{this[row, column]} ");
+                    }
+                }
+            }
+        }
+        //5.
+        public int Track()
+        {
+            int sum = DiagonalMatrixArray.Sum();
+            return sum;
+        }
+        //6.
+        public override string ToString()
+        {
+            return string.Join(", ", DiagonalMatrixArray);
+        }
+        public override bool Equals(object obj)
+        {
+            //according to Jeffrey Richter (CLR via C#)
+            var item = obj as Matrix;
+            if (item == null) //if obj args is null return false
+            {
+                return false;
+            }
```

Comment:
No need to do this, if item is not Matrix, you will return false in line 111 after check for null after casting